### DISCUSSION TITLE
Poll: What's your expected behavior of set_fields and --set?

### DISCUSSION THREAD

**Discussion by bertbesser** [beetbox/beets#3932] [General]
Hi everybody,

we are discussing the expected behavior of the [set_fields configuration](https://beets.readthedocs.io/en/stable/reference/config.html?highlight=set_fields#set-fields), which uses the `set_fields` configuration file block and the `--set` command line option. #3927 

*Question for you: Should fields be set on the imported album **only**, or also on each individual track?*

In the current implementation, fields are set on the imported *album only*. Hence `beet list` won't let you filter for those fields unless you provide the `-a` flag. With `-a`, however, you will see a list of albums instead of individual tracks.

There are some discussions on the board (e.g. [here](https://discourse.beets.io/t/plugin-auto-set-comment-on-import/1762) or [here](https://discourse.beets.io/t/set-track-field-on-import/1235/5) or [here](https://discourse.beets.io/t/set-field-to-url/418/7)) and on the internet (e.g. [here](https://github.com/beetbox/beets/issues/2800)) where people expect that fields are also set *on each individual track*. In particular, this would imply that the fields are also written to the metadata of the track files instead of only beets' database.

*Main question: Would changing the implementation from **album in database only** to **write each individual track file** break your setup?*

Please share your thoughts.
Thank you!

PS: Also posted on [discourse](https://discourse.beets.io/t/poll-whats-your-expected-behavior-set-fields-and-set/1764).

**Comment by teogramm** [comment]
For me the main issue is that if fields are not set on each individual track, they are not detected in paths configuration. For example I used to have a configuration like this:
```
paths:
    default: ....
    custom:1: custom_folder/......
    custom2:1: custom2/.....
```
But this meant that I could not simply do ``beet import --set custom=1`` but instead had to run ``beet modify custom=1 ...`` after importing each album which got pretty tiring. I now use a configuration like this one: https://github.com/beetbox/beets/issues/2797#issuecomment-669476875 and it works pretty well.

Anyway, a way to change this is to maybe split the ``set_fields`` into ``set_album_fields`` and ``set_track_fields`` while making ``set_fields`` set both fields. People relying on the old behaviour could then change ``set_fields`` to ``set_album_fields``. ( I do not know how easy or hard this would technically be).

**Comment by rnkn** [comment]
(Duplicated from Discourse...)
I assumed it tried to set the values on each individual track and when it didn't seem to work I always just thought this was a bug.
